pokemongo_bot/cell_workers/pokemon_catch_worker.py

```diff
@@ -178,6 +178,8 @@ def _pct(self, rate_by_ball):
         return '{0:.2f}'.format(rate_by_ball * 100)
 
     def _use_berry(self, berry_id, berry_count, encounter_id, catch_rate_by_ball, current_ball):
+        # Delay to simulate selecting berry
+        action_delay(self.config.action_wait_min, self.config.action_wait_max)
```

Comment:
Selecting a berry doesn’t make a request to the servers. It’s handled in the app. 
When you want to select a berry in the app you open your inventory. I guess this trigger a call to update the inventory. Then you select a berry, but Niantic doesn’t know about it. It knows only when you throw it on the pokemon.